### ISSUE TITLE
Spyder 4.1.5 - disable warnings

### ISSUE THREAD

**Issue by andrea83ny** [spyder-ide/spyder#14313]
There is currently no way to disable warnings (apparently).
I'm going nuts, I have an application with thousands rows of code and Spyder prints million of rows because of useless warnings.
Using simple python syntax this should fix the problem but it doesn't:
warnings.filterwarnings("ignore")

Can you please fix your application?
Thanks

**Comment by andrea83ny** [comment]
https://xbuba.com/questions/46568959
"Tools > Preferences > Editor > Code Introspection/Analysis"
This doesn't exist anymore.
Now, can somebody add something in plain English called "Enable/Disable warnings"?

**Comment by dalthviz** [comment]
Hi @andrea83ny to disable the warnings you can go to the `Source` menu and uncheck `Show code style warnings` and `Show docstring style warnings`. Also, there is an option in the `Preferences` dialog to disable basic linting of the files. For that go to `Tools > Preferences > Completion and linting > Linting` and uncheck `Enable basic linting`:

### Source menu

![source](https://user-images.githubusercontent.com/16781833/100272275-d2658d00-2f28-11eb-8d14-d895c0e9e29e.png)


### Preferences dialog

![image](https://user-images.githubusercontent.com/16781833/100272134-9df1d100-2f28-11eb-81cd-13ef39d5edec.png)

Hope this info helps navigate the config available :)

**Comment by andrea83ny** [comment]
Under "Source" the only option I have enabled is "Show code folding". Hence, the expectation is that if "Shoe code style warnings" is disabled that no warnings should be displayed in the console. Right?
I also unchecked the option "Enable basic lintining" and I still get all the warnings.
Thanks in advance for your help

**Comment by dalthviz** [comment]
Oh I misunderstood, you are trying to disable warnings in the IPython Console (the config I gave is for warnings in the Editor). I think that using 

```
import warnings
warnings.filterwarnings("ignore")
```

should work as you say. Are you running those lines directly on the IPython Console or at the start of the code you are running? @impact maybe you can have some suggestions regarding this one?

**Comment by andrea83ny** [comment]
I have a python application that imports different other modules I wrote.
However, the very first thing the script does is to import os, import sys, and that's also where I have

> import warnings
> warnings.filterwarnings("ignore")

however, I keep getting all the warnings inside the console :(

**Comment by dalthviz** [comment]
Could you give us an example code snippet to reproduce this @andrea83ny ? I'm trying running this (which prints a `DepreciationWarning`):

```
import numpy as np
np.random.random_integers(0, 1000, 1000)
```

but after I add the lines to suppress warnings the message is not printed anymore:

![warnings](https://user-images.githubusercontent.com/16781833/100278098-49ebea00-2f32-11eb-844a-f7c46aefe396.gif)

**Comment by impact27** [comment]
> 
> 
> I have a python application that imports different other modules I wrote.
> However, the very first thing the script does is to import os, import sys, and that's also where I have
> 
> > import warnings
> > warnings.filterwarnings("ignore")
> 
> however, I keep getting all the warnings inside the console :(

I can think of two explainations:
 - Some module you are importing is resetting the warning filter
 - What you are seeing are not "real" python warnings. What I mean is that if you do:
```
print("Warning: something went wrong")
```
This is not a warning but a print, as far as python is concerned.

Could you give a reproducible short example where `warnings.filterwarnings("ignore")` doesn't disables warnings? Without that we will not be able to help you.

**Comment by andrea83ny** [comment]
You're actually write, there were some other modules that later in the code were changing the status of the warnings filter (in my case such module was cvxpy). Thanks for your help, much appreciated. I'm a big fan and full time user of Spyder!

**Comment by dalthviz** [comment]
Glad you figure it out @andrea83ny ! and thanks for the feedback here @impact much appreciated :)

Closing this one then
